### PR TITLE
Add support for writing test cases directly in WAT.

### LINKED ISSUE #130: Add support for running direct WASM / WAT tests directly
Presently, the built and test system assumes that all of the tests are based on Rust code, and live in an appropriate subtree of `test-fixtures`. We should extend this to support hand-written WASM tests, as well, perhaps as a fore-runner to having other languages inside the test suite as well.

### PR DESCRIPTION
**Pull request by acw**
This version leverages `wasmtime`'s ability to compile/run WAT directly to avoid the need to install additional tooling on the build machines, at the cost of requiring some fiddling with the test harness.

Resolves #130.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -29,7 +31,7 @@ jobs:
       uses: actions/cache@v1
       with:
         path: target
-        key: ${{ runner.os }}-cargo-build-target-v1-${{ hashFiles('**/Cargo.lock') }}
+        key: ${{ runner.os }}-cargo-build-target-v${{ env.CACHE_GENERATION }}-${{ hashFiles('**/Cargo.lock') }}
```
Comment (fgsch) [inline .github/workflows/test.yml:34]: Perhaps something to do in a different PR but should this be extended to the other key in that file?

Comment (acw) [reply .github/workflows/test.yml:34]: I wasn't sure. It'd be easy enough to do, but I admit I don't have much depth when it comes to these caching mechanisms. So I limited to just the key that seemed to be failing.

Do you understand these things? I'm happy to push a patch if you do and think it'd be a good idea. If not, any idea who does?

Comment (fgsch) [reply .github/workflows/test.yml:34]: To some extent :) I think we should cover both. They have the same key (based on the output from the Github Actions) so in theory the jobs can share the files. If they can't (because the keys end up being different) and we need to bump the CACHE_GENERATION value, worse case the first time the job runs it will need to download and build everything. For reference, this will happen naturally if the cache file is not accessed in 7 days.

Comment (acw) [reply .github/workflows/test.yml:34]: Done! (In 2b58192)

**Review by cratelyn — COMMENTED** [review]
this looks great! i love how this new test plumbing looks, and _really_ appreciate a WAT example that we can build other tests from. i caught a tiny comment nitpick related to a previously broken link, but the contents of this PR look good to me. :sparkles:

**Review by cratelyn — APPROVED** [review]
who, WAT, when, where, why :sparkles: :heavy_check_mark:
